src/shared/bus-util.h

```diff
@@ -23,11 +23,31 @@ typedef enum BusTransport {
 } BusTransport;
 
 typedef struct BusLocator {
-        const char    *destination;
-        const char    *path;
-        const char    *interface;
+        const char *destination;
+        const char *path;
+        const char *interface;
 } BusLocator;
 
+typedef struct BusObjectImplementation BusObjectImplementation;
+
+typedef struct BusObjectVtablePair {
+        const sd_bus_vtable *vtable;
+        sd_bus_object_find_t object_find;
+} BusObjectVtablePair;
+
+struct BusObjectImplementation {
+        const char *path;
+        const char *interface;
+        const BusObjectVtablePair *vtables;
+        sd_bus_node_enumerator_t node_enumerator;
+        bool manager;
+        const BusObjectImplementation* children[];
+};
+
+#define VTABLE_LIST(...) ((const BusObjectVtablePair[]) { __VA_ARGS__, {} })
+#define CHILD_LIST(...) { __VA_ARGS__ __VA_OPT__(,) NULL }
+#define IMPLEMENTATION_LIST(...) ((const BusObjectImplementation* []) { __VA_ARGS__, NULL })
```

Comment:
hmm, where is this one used?

Reply:
In the `--bus-introspect` implementations.